src/coreclr/System.Private.CoreLib/src/System/Delegate.CoreCLR.cs

```diff
@@ -56,6 +57,7 @@ protected Delegate(object target, string method)
         // This constructor is called from a class to generate a
         // delegate based upon a static method name and the Type object
         // for the class defining the method.
+        [RequiresUnreferencedCode("The target method might be removed")]
```

Comment:
This could be better fixed by adding `[DynamicallyAccessedMembers(DynamicallyAccessedMemberType.PublicMethods | DynamicallyAccessedMemberType.NonPublicMethods)]`, right?

Reply:
True. Fixing.

Reply:
I will fix it that way. Note that I need to suppress the warning calling `BindToMethodName`, which is annotated as `RequiresUnreferencedCode`.

Reply:
Maybe we should fix that as well - @MichalStrehovsky: I don't see anything wrong with changing the annotation BindToMethodName to add [DAM(AllMethods)] on the methodType....

Reply:
I don't mind either way - this constructor and the one on MulticastDelegate is pretty much dead code - I'm not aware of any language compiler generating calls to this constructor from delegate type definitions they generate. It's probably only reachable from IL.